client/src/components/Dataset/DatasetList.vue

```diff
@@ -153,7 +153,17 @@ export default {
         async onShowDataset(item) {
             const historyId = item.history_id;
             try {
-                await this.applyFilterText(historyId, `hid:${item.hid}`);
+                let filterText = "";
+                if (item.hid) {
+                    filterText = `hid:${item.hid}`;
+                    if (item.deleted) {
+                        filterText += " deleted:true";
+                    }
+                    if (!item.visible) {
+                        filterText += " visible:false";
+                    }
+                }
+                await this.applyFilterText(historyId, filterText);
```

Comment:
Not directly related to this PR, but I have the feeling that exposing `applyFilterText` forces every component that uses it to know how to build a search string. 
Maybe exposing `applyFilter` instead and passing a filter object would be better. Something like:

```ts
const filter = {
    hid: item.hid,
    deleted: item.deleted,
    visible: !item.visible,
}
await this.applyFilter(filter);
```

This will also allow defining an interface for the filters object in typescript and will make building the search text less manual and brittle:

```ts
interface HistoryFilters {
    hid?: number;
    deleted?: boolean;
    visible?: boolean;
    related?: number;
    ...
}
```
```

Reply:
Hey @davelopez, I've implemented the usage of the `Filtering` class in places where we manually update the `filterText` in the latest commit.

Reply:
Thank you @ahmedhamidawan! this is already a significant improvement! :+1: 

In the long run, I still think it would be better to work consistently with the filters as an object (strongly typed with an interface) instead of the "filter text" or a generic `Record<string, string | boolean>`. The filter text should only be the way of representing the filter in the UI input box and the filter object will be the internal way of working, storing, and maintaining filters.

But this probably would require a bigger refactoring and maybe probably worth doing as a separate PR if at all :)

Reply:
Thanks @davelopez ! I'll soon be pushing a commit that has a `type FilterVal = string | boolean;` replacing all occurrences of `<T>` and `string | boolean`. But yes, if we could restrict the usage of `filterText` to only the text fields, and not as a value passed to the store etc. it would be nicer, and can be done in another refactor PR.